projects/sel4/porting.md

```diff
@@ -7,79 +7,159 @@ SPDX-FileCopyrightText: 2020 seL4 Project a Series of LF Projects, LLC.
 
-### CMake-build system
+## Testing your seL4 port
 
-You will need to modify the build system in two places. Firstly in the kernel's platform folder, you define `kernel/src/plat/<platform>/config.cmake`, again you can look at other platforms for examples.
-Secondly, you will need to add 'KernelPlatform' to the cmake and elf-loader tools. You can do this by editing
-`tools/cmake-tool/helpers/application_settings.cmake` and `tools/elfloader-tool/CMakeLists.txt`.
+In order to sanity check your port of seL4, it is a good idea to get
+[the seL4test project](https://docs.sel4.systems/projects/sel4test/) running and passing.
 
 ### ELF-loader
 
-You will need to add directories and files `tools/elfloader-tool/src/plat/<platform>/sys_fputc.c` and
-`tools/elfloader-tool/include/plat/<platform>/platform.h` which will provide a simple implementation of putchar with a physical address for uart for the elf loader to use.
+Before we can run sel4test, we need to have seL4 booting first.
 
-## seL4 test
+Although we use U-Boot as a bootloader, there are additional steps needed to initialise the
+hardware and setup system images before we can boot seL4. This is the purpose of the
+[ELF-loader](https://docs.sel4.systems/projects/elfloader/).
 
-seL4test needs user level serial and timer drivers to run. Add these drivers and the following files to libplatsupport in
-order to get the test suite running. The directory structure and files will look something like the following. Some of these
-files may not have anything in them but include them anyway to keep the build system happy.
+The code for the ELF-loader can be found in the `elfloader-tool` directory at the root of
+the [seL4_tools](https://github.com/seL4/seL4_tools) repository.
 
-Use the other platforms as examples; generally you will have to configure the timer driver by reading the manual, it may have its
-own unique programming sequence and can sometimes be a bit tricky.
+Depending on your platform, you may have to make changes to the ELF-loader which is
+responsible for loading the kernel and any user-programs. The ELF-loader outputs to the
+serial, so needs to be able to access the UART.
 
-You can find the physical addresses and interrupt numbers for each device in its platform reference manual and copy them into the right places.
+On RISC-V, outputting to UART is done via a call to the SBI layer so no changes are necessary.
 
-`libplatsupport/plat_include/<platform>/`
-	-`clock.h`
-	-`gpio.h`
-	-`i2c.h`
-	-`mux.h`
-	-`serial.h`
-	-`timer.h`
-`libplatsupport/src/plat/<platform>/`
-	-`chardev.c`
-	-`ltimer.c`
-	-`serial.c`
-	-`timer.c`
+On ARM, there are a variety of existing UART drivers in `tools/elfloader-tool/src/drivers/uart/`,
+if the default serial is not supported by any of these drivers, you will have to
+provide a simple putchar implementation.
+
+If the ELF-loader needs to be a particular file format (e.g binary or EFI) or needs to start at a
```

Comment:
needs to have?

Reply:
I think "needs to be" is correct? But I've changed this sentence to make it clearer now.